src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -260,6 +261,16 @@ internal virtual ApplicationConfiguration BuildConfiguration()
                 throw new InvalidOperationException(CoreErrorMessages.AdfsNotCurrentlySupportedAuthorityType);
             }
 
+            if (string.IsNullOrWhiteSpace(Config.RedirectUri))
+            {
+                Config.RedirectUri = Constants.DefaultRedirectUri;
```

Comment:
>Constants.DefaultRedirectUri [](start = 37, length = 28)

this `urn:ietf:wg:oauth:2.0:oob` won't work w/system browser on iOS or Android. we already have a check to see if this redirect URI is being used (i'm running into that issue now) in Validate(), but should we be setting it here if they don't pass something in? will that be confusing?

Reply:
If the user doesn't set an authority, we set a default.  I think that makes sense for the redirecturi as well.  It should _not_ be Constants.DefaultRedirectUri however, you are very correct here.  I'm updating this to get the value from PlatformProxyFactory to get the platform specific default if one is not set.  for the scenarios that don't need it, then it will be set and unused.

---
In reply to: [250033584](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/822#discussion_r250033584) [](ancestors = 250033584)